vue-components/src/components/Lookup.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+	<label :class="[ 'wikit', 'wikit-Lookup', `wikit-Lookup--${width}` ]">
+		<div class="wikit-Lookup__label">{{ label }}</div>
+		<Input
+			@input.native="inputValue = $event.target.value"
+			@focus.native="focused = true"
+			@blur.native="focused = false"
+			:feedback-type="feedbackType"
+			:placeholder="placeholder"
+			:disabled="disabled"
+		/>
+		<LookupMenu :menu-items="menuItems" v-if="showsMenu" />
+		<ValidationMessage
+			v-if="error"
+			:type="error.type"
+			:message="error.message"
+		/>
+	</label>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import ValidationMessage from './ValidationMessage.vue';
+import Input from './Input.vue';
+import LookupMenu from './LookupMenu.vue';
+
+/**
+ * Text input fields are form elements that let users input and edit values in the form of text.
+ *
+ * Uses the following components internally: Input, ValidationMessage
+ */
+export default Vue.extend( {
+	name: 'Lookup',
+	data() {
+		return {
+			focused: false,
+			inputValue: '',
+		};
+	},
+	props: {
+		menuItems: {
+			type: Array,
+			default: (): [] => [],
```

Comment:
`[]` literally means empty array, which is technically correct for the default?! I'd have expected this to eventually declare it to be a list of menu items, though.

Reply:
we discussed it, we think it is fine for now.